components/brave_wallet/browser/zcash/zcash_wallet_service_tasks.h

```diff
@@ -0,0 +1,119 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ZCASH_ZCASH_WALLET_SERVICE_TASKS_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_ZCASH_ZCASH_WALLET_SERVICE_TASKS_H_
+
+#include <set>
+#include <string>
+#include <vector>
+
+#include "brave/components/brave_wallet/browser/zcash/zcash_wallet_service.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+
+namespace brave_wallet {
+
+class GetTransparentUtxosContext
+    : public base::RefCountedThreadSafe<GetTransparentUtxosContext> {
```

Comment:
why are these RefCountedThreadSafe? Use of ref-counting should be limited to cases where something is truly shared and also this class does not appear to be thread safe 
https://www.chromium.org/developers/coding-style/important-abstractions-and-data-structures/#scoped_refptrt-baserefcounted-baserefcountedthreadsafe